src/pages/components/Overlay.qml

```diff
@@ -415,15 +442,23 @@ PanelBackground {
 
             onStatusChanged: {
                 if (activeWebPage && status == PageStatus.Active) {
-                    activeWebPage.grabToFile()
+                    webView.privateMode ? activeWebPage.grabThumbnail() : activeWebPage.grabToFile()
                 }
             }
 
             Browser.TabView {
                 model: webView.tabModel
                 portrait: tabPage.isPortrait
+                privateMode: webView.privateMode
 
                 onHide: pageStack.pop()
+
+                onPrivateModeChanged: {
+                    webView.privateMode = privateMode;
```

Comment:
There is a belief that explicit statement separator (the semicolons) do cognitive overload with information noise and thus make developers less productive. The majority of SailfishOS developers are the believers so we'd better not use semicolons in QML for the sake of consistency.